node/src/store/in_memory_store.rs

```diff
@@ -301,6 +306,35 @@ impl InMemoryStoreInner {
 
         Ok(Some(metadata.clone()))
     }
+
+    fn remove_last(&mut self) -> Result<u64> {
```

Comment:
I think instead of returning `u64`, which many never be used, it should be nice to return `(ExtendedHeader, SamplingMetadata)`. That way you can use CIDs of sampling metadata without having a possibility of get/set/remove race.